src/core/trulens/core/guardrails/base.py

```diff
@@ -98,4 +101,168 @@ def wrapper(*args, **kwargs):
+                "Specify `keyword_for_prompt` to avoid this warning."
+            )
+
+        def wrapper(*args, **kwargs):
+            bindings = sig.bind(*args, **kwargs)
+            keyword_value = bindings.arguments[self.keyword_for_prompt]
+            result = self.feedback(keyword_value)
+            if not isinstance(result, float):
+                raise ValueError(
+                    "`block_input` can only be used with feedback functions that return a float."
+                )
+            if (self.feedback.higher_is_better and result < self.threshold) or (
+                not self.feedback.higher_is_better and result > self.threshold
+            ):
+                return self.return_value
+
+            return func(*args, **kwargs)
+
+        # note: the following information is manually written to the wrapper because @functools.wraps(func) causes breaking of the method.
+        wrapper.__name__ = func.__name__
+        wrapper.__doc__ = func.__doc__
+        wrapper.__signature__ = sig
+        return wrapper
+
+
+class block_output:
+    """Provides a decorator to block output based on a given feedback and threshold.
+
+    Args:
+        feedback: The feedback object to use for blocking. It must only take a single argument.
+        threshold: The minimum feedback value required for a context to be included.
+        return_value: The value to return if the input is blocked. Defaults to None.
+
+    Example:
+        ```python
+        from trulens.core.guardrails.base import block_output
+
+        feedback = Feedback(provider.criminality, higher_is_better = False)
+
+        class safe_output_chat_app:
+            @instrument
+            @block_output(feedback = feedback,
+                threshold = 0.5,
+                return_value = "Sorry, I couldn't find an answer to your question.")
+            def chat(self, question: str) -> str:
+                completion = (
+                    oai_client.chat.completions.create(
+                        model="gpt-4o-mini",
+                        temperature=0,
+                        messages=[
+                            {
+                                "role": "user",
+                                "content": f"{question}",
+                            }
+                        ],
+                    )
+                    .choices[0]
+                    .message.content
+                )
+                return completion
+        ```
+    """
+
+    def __init__(
+        self,
+        feedback: core_feedback.Feedback,
+        threshold: float,
+        return_value: Optional[str] = None,
+    ):
+        self.feedback = feedback
+        self.threshold = threshold
+        self.return_value = return_value
+
+    def __call__(self, func):
+        sig = inspect.signature(func)
+
+        def wrapper(*args, **kwargs):
+            output = func(*args, **kwargs)
+            result = self.feedback(output)
+            if not isinstance(result, float):
```

Comment:
will this work with feedback results like 
{0.5, 'reasons': "COT reasons str..."}?

Reply:
No, must be score-only feedback. That disclaimer is in the docs and there is a dedicated error message for that case. We could allow reasons feedback but that would be extra tokens generated that are not seen/used, so it seems better to not allow it

Reply:
Could also generalize to allow feedbacks with selectors. Perhaps add a jira issue for future work?

Reply:
I think this does allow feedbacks with selectors, but I'll create a JIRA nonetheless

Reply:
Would be good also to allow feedbacks that take multiple args, such as a way to block ungrounded responses